lib/gui/app/components/source-selector/source-selector.tsx

```diff
@@ -164,20 +166,21 @@ const URLSelector = ({
 					}
 				/>
 			</Flex>
-			{!_.isEmpty(recentImages) && (
+			{recentImages.length > 0 && (
 				<Flex flexDirection="column">
 					<Txt fontSize={18}>Recent</Txt>
 					<Card
 						style={{ padding: '10px 15px' }}
-						rows={_.map(recentImages, (recent) => (
+						rows={recentImages.map((recent) => (
 							<Txt
 								key={recent}
 								onClick={() => {
 									setImageURL(recent);
 								}}
 							>
 								<span>
-									{_.last(_.split(recent, '/'))} - {recent}
+									{middleEllipsis(recent.split('/').pop() || '', 45)} -{' '}
+									{middleEllipsis(recent, 45)}
```

Comment:
If the goal was to avoid a scroll bar, I still get a scroll bar

![Capture d’écran de 2020-07-03 12-12-26](https://user-images.githubusercontent.com/180331/86459100-96861d00-bd26-11ea-9797-458791383097.png)

Reply:
I'm not sure we should limit the url length, just let it show on multiple lines.
You don't know if `middleEllipsis` will make it short enough:

![Capture d’écran de 2020-07-03 12-20-48](https://user-images.githubusercontent.com/180331/86459943-d00b5800-bd27-11ea-90b3-0d870c497b33.png)